broker-node/app/Http/Controllers/UploadSessionController.php

```diff
@@ -77,7 +85,21 @@ public function edit($id)
      */
     public function update(Request $request, $id)
     {
-        //
+        $session = UploadSession::find($id);
+        if (empty($session)) return response('Session not found.', 404);
+
+        $genesis_hash = $session['genesis_hash'];
+        $chunk = $request->input('chunk');
+
+        $data_mapping = DataMapping::where('genesis_hash', $genesis_hash)
+            ->where('chunk_idx', $chunk['idx'])
+            ->first();
+        if (empty($data_mapping)) return response('Datamap not found', 404);
+
+        // TODO: What to do with $data_mapping['hash']
```

Comment:
@pandora23 Once I have the datamap, what needs to be done with the hash?

Reply:
i think we need to make sure $data_mapping['hash'] == $chunk['hash'] to verify that they're coming from the same source?